src/Symfony/Component/RateLimiter/LimiterFactory.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+use Symfony\Component\Lock\LockFactory;
+use Symfony\Component\Lock\NoLock;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\RateLimiter\Storage\StorageInterface;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @final
+ */
+class LimiterFactory
```

Comment:
Can we rename it to `Limiter`?

Reply:
Hmm, I'm not completely sure about this. At the moment, I've used the same tactics as the Lock component (which also has a factory + limiters) and register 2 autowirings for each limiter:

* `LimiterInterface $<limiterName>Limiter`, which injects `$limiterFactory->createLimiter()` (i.e. an application wide limiter)
* `LimiterFactory $<limiterName>LimiterFactory`, which injects `$limiterFactory`, so a class can define a specific limiter key (see the [login throttling implementation](https://github.com/symfony/symfony/pull/38204/files#diff-e0d15bf4273820e4a8f71be498797fc7R49-R51))

If we call this one `Limiter`, you will get these autowiring config:

* `LimiterInterface $<limiterName>Limiter`
* `Limiter $<limiterName>Limiter`

Or should I instead also drop the application-wide limiter autowiring (so a user would always need to do `$limiter->create()` or `$limiter->create(...)`)?

Reply:
I'm not sure that the application-wide limiter autowiring is that useful, but I might be wrong.

Reply:
the other issue we have is that we have a `Limiter` class which is then not an implementation of the `LimiterInterface`, which can be quite confusing.